### PR TITLE
vulkan-headers: define package_id_embed_mode

### PR DESCRIPTION
**Pull request by jcar87**
Specify library name and version:  **vulkan-headers/all**

Conan 2.0 has new logic when embedding the information of dependencies in the package info (for the package ID). 
When a consumer depends on a package type of `header-library`, the package info of that downstream consumer will contain the entire reference of the package used from the dependency  - this will include name, version, recipe revision, package ID. This means that any changes in the recipe revision of the dependency (when resolving the graph), would yield a completely new package ID, and thus a rebuild. This is conservative, as it assumes that a change in a header only library actually changes the contents (or its behaviour), and thus affects the downstream consumers.

We can probably relax this in Conan Center for header only libraries, under the assumption that _for the same version_ the contents do not change. But this also means that we have to be more judicious when reviewing changes to recipes of header-only libraries.  Also serves as a good motivator to document this attribute, which can affect how consumers can be affected to changes, and is meant to allow the recipe/library authors to reflect how the versioning scheme is done (rather than impose this on consumers, or always assume semver implicitly).

### CODE REVIEW

File: recipes/vulkan-headers/all/conanfile.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ class VulkanHeadersConan(ConanFile):
     homepage = "https://github.com/KhronosGroup/Vulkan-Headers"
     url = "https://github.com/conan-io/conan-center-index"
     package_type = "header-library"
+    package_id_embed_mode = "patch_mode"
```
Comment (SpaceIm) [inline recipes/vulkan-headers/all/conanfile.py:17]: I don't see any reference to `package_id_embed_mode` attribute in conan v2 documentation, and even less possible values and their meaning.

**Comment by paulharris** [comment]
I don't think I like the assumption that header-only libraries won't change if the recipe changes.
Recipes include patches and replace_into_file... That would clearly change the header file contents...
You have "os" "compiler" etc defined in the recipe's package id mix, why not add "patches" IF that recipe ever had patches?
So simple recipes without patches will match your loose assumptions, but more complex recipes will not spoil the system?

**Comment by SpaceIm** [comment]
@jcar87 invalidation of latest pre-built package of `vulkan-loader` comes from https://github.com/conan-io/conan-center-index/pull/16886 (which is good, now RREV & PREV of xorg recipe don't contribute to package id of downstream recipes), not vulkan-header (it could, but it's not what happened here).

**Review by SSE4 — CHANGES_REQUESTED** [review]
first off, recipes shouldn't use undocumented features. right now, `package_id_embed_mode` hits zero matches in conan docs. per [docs](https://docs.conan.io/2/introduction.html) `Only documented features in https://docs.conan.io/ are considered part of the public interface of Conan. Private implementation details, and everything not included in the documentation is subject to change.`. so if something is not documented, it could be changed and removed any time and shouldn't be used in ConanCenter recipes, and this change cannot be accepted (we cannot advertise to users to rely on that attribute).

second, as correctly specified by @paulharris, changes in recipe may imply header file contents change. not just patches or `replace_in_file`, but simple things like `copy` in`package` method may affect on headers layout, for instance. therefore, default mode should stay regardless.

and last, it seems like problem doesn't come from `vulkan-headers` at all per @SpaceIm .

File: recipes/vulkan-headers/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/vulkan-headers/all/conanfile.py:17]: @SpaceIm it's documented now: https://docs.conan.io/2/reference/conanfile/attributes.html#package-id-embed-non-embed-unknown-mode

**Review by memsharded — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**